project_task_description_template/models/project_task.py

```diff
@@ -13,6 +13,14 @@ class ProjectTask(models.Model):
 
     @api.onchange("description_template_id")
     def _onchange_description_template_id(self):
-        if self.description_template_id:
-            description = self.description if self.description else ""
-            self.description = description + self.description_template_id.description
+        for task in self:
+            if not task.description_template_id:
+                continue
+            template_text = task.description_template_id.description or ""
+            if not template_text:
+                continue
+            current = task.description or ""
+            # Avoid duplicating the same template content
+            if current == template_text or current.endswith(template_text):
```

Comment:
Checkout my regex solution on v14.  it covers the template present in any part of the text.
https://github.com/OCA/project/pull/1585/commits/32a77d63761bf9faef4b50cd337ef9b97fc494af